src/clr-angular/button/button-loading/loading-button.ts

```diff
@@ -5,26 +5,84 @@
  */
 
 
-import {Component} from "@angular/core";
-import {LoadingListener} from "../../utils/loading/loading-listener";
-
+import {animate, keyframes, style, transition, trigger} from "@angular/animations";
+import {Component, ElementRef, EventEmitter, Output, Renderer2} from "@angular/core";
+import {ClrLoadingState, LoadingListener} from "../../utils/loading";
 
 @Component({
     selector: "button[clrLoading]",
     template: `
-        <span class="spinner spinner-inline" *ngIf="loading"></span>
-        <ng-content></ng-content>
+        <ng-container [ngSwitch]="state">
+            <span *ngSwitchCase="buttonState.LOADING" class="spinner-btn-wrapper">
```

Comment:
Did the switch solve your animation issue?

Reply:
Nope, unfortunately the same results as using `*ngIf`.